File: app/src/main/java/com/tuvakov/zeyoube/android/dagger/ExecutorModule.java
package com.tuvakov.zeyoube.android.dagger;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
abstract class ExecutorModule {
    @Provides @Singleton
    static ExecutorService providesSingleThreadExecutor() {
        return Executors.newSingleThreadExecutor();
    }
}

File: app/src/main/java/com/tuvakov/zetube/android/dagger/ExecutorModule.kt
package com.tuvakov.zetube.android.dagger

import dagger.Module
import dagger.Provides
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors
import javax.inject.Singleton

@Module
internal object ExecutorModule {
    @JvmStatic
    @Provides
    @Singleton
    fun providesSingleThreadExecutor(): ExecutorService {
        return Executors.newSingleThreadExecutor()
    }
}